### ISSUE TITLE
Add option to disable reset button in `gr.Slider`

### ISSUE THREAD

**Issue by JackismyShephard** [gradio-app/gradio#9973]
- [x] I have searched to see if a similar issue already exists.

I notice that a reset button has been added to slider components. That is nice, but what would be even better if you could add reset buttons for other components as well. For my workflow I specifically need reset buttons for (single-select) dropdown components and radio components. 

**Is your feature request related to a problem? Please describe.**  

**Describe the solution you'd like**  

**Additional context**

**Comment by abidlabs** [comment]
Hi @JackismyShephard I don't think we'll add these buttons as they would clutter the UI too much, but I think this would be a great case for a custom component!

tldr: we've made it possible for Gradio users to create their own custom components -- meaning that you can write some Python and JavaScript (Svelte), and publish it as a Gradio component. You can use it in your own Gradio apps, or share it so that anyone can use it in their Gradio apps. 

Here's an example of a Gradio custom component: https://github.com/PhyscalX/gradio-image-prompter -- it lets you upload images and process points or draw boxes on top of the image.

If you'd like to create your own, you could template off of the dropdown and radio components, we've put together a Guide: https://www.gradio.app/guides/five-minute-guide, and we're happy to help.

**Comment by JackismyShephard** [comment]
@abidlabs I actually agree with you. Could you then perhaps make it possible to disable the buttons on the slider components (via an initialization option) because they also clutter and I need a global reset button anyways

**Comment by abidlabs** [comment]
We could probably do that, let me rename the issue to reflect that

**Comment by abidlabs** [comment]
Btw if you'd like to make a PR to that effect (adding a `show_reset_button` parameter to `gr.Slider`, happy to review)

**Comment by JackismyShephard** [comment]
@abidlabs I think I could try and take a swing at that. I probably need some pointers from you though . Primarily in terms of where in your code base I need to make changes

**Comment by abidlabs** [comment]
Hi @JackismyShephard if you are still interested, the relevant sections of the code would be:

* [here in the frontend](https://github.com/gradio-app/gradio/blob/2afcad80abd489111e47cf586a2a8221cc3dc9b6/js/slider/Index.svelte#L127) 
* [here in the backend](https://github.com/gradio-app/gradio/blob/6f95286337459efbccb95c9cfac63355669df9ee/gradio/components/slider.py#L30)

**Comment by amanchauhan11** [comment]
Hi @abidlabs if no one else is working on this, I would like to make a PR for it.

**Comment by abidlabs** [comment]
Sure feel free to open a PR @amanchauhan11

**Comment by JackismyShephard** [comment]
@amanchauhan11 very nice !
